web/packages/teleport/src/Console/DocumentSsh/useFileTransfer.ts

```diff
@@ -14,88 +14,47 @@
  * limitations under the License.
  */
 
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
+import { useFileTransferContext } from 'shared/components/FileTransfer';
 import { FileTransferRequest } from 'shared/components/FileTransfer/FileTransferRequests';
-import { useFilesStore } from 'shared/components/FileTransfer/useFilesStore';
 
-import { UserContext } from 'teleport/services/user';
+import Tty from 'teleport/lib/term/tty';
+import { EventType } from 'teleport/lib/term/enums';
+import { Session } from 'teleport/services/session';
+import { DocumentSsh } from 'teleport/Console/stores';
 
-import { DocumentSsh } from '../stores';
+import { useConsoleContext } from '../consoleContextProvider';
 
 import { getHttpFileTransferHandlers } from './httpFileTransferHandlers';
 import useGetScpUrl from './useGetScpUrl';
 
-export const useFileTransfer = ({ doc, user, addMfaToScpUrls }: Props) => {
-  const filesStore = useFilesStore();
+const useFileTransfer = (
+  tty: Tty,
+  session: Session,
+  currentDoc: DocumentSsh,
+  addMfaToScpUrls: boolean
+) => {
+  const { filesStore } = useFileTransferContext();
+  const ctx = useConsoleContext();
+  const user = ctx.getStoreUser();
   const [fileTransferRequests, setFileTransferRequests] = useState<
     FileTransferRequest[]
   >([]);
   const { getScpUrl, attempt: getMfaResponseAttempt } =
     useGetScpUrl(addMfaToScpUrls);
 
-  function updateFileTransferRequests(data: FileTransferRequest) {
-    const newFileTransferRequest: FileTransferRequest = {
-      ...data,
-      isOwnRequest: user?.username === data.requester,
-    };
-    return setFileTransferRequests(prevstate => {
-      // We receive the same data type when a file transfer request is created and
-      // when an update event happens. Check if we already have this request in our list. If not
-      // in our list, we add it
-      const foundRequest = prevstate.find(
-        ft => ft.requestID === newFileTransferRequest.requestID
+  useEffect(() => {
```

Comment:
[React 18 will call hooks twice by default in dev mode](https://react.dev/reference/react/useEffect#my-effect-runs-twice-when-the-component-mounts), so I think it'd make sense to future-proof this hook by returning a cleanup function which removes these three specific event listeners that are added on `tty`.